labelbox_dev/task.py

```diff
@@ -0,0 +1,186 @@
+import json
+import logging
+import time
+from abc import ABC, abstractmethod
+from enum import Enum
+from functools import lru_cache
+from typing import Any, Dict, Union
+
+import requests
+
+from labelbox_dev.entity import Entity
+from labelbox_dev.exceptions import ResourceNotFoundError
+from labelbox_dev.session import Session
+from labelbox_dev.utils import format_json_to_snake_case
+
+TASK_RESOURCE = "tasks"
+
+logger = logging.getLogger(__name__)
+
+
+class TaskStatus(Enum):
+    COMPLETE = "COMPLETE"
```

Comment:
Should it be `COMPLETED`?

Reply:
Task status in the backend is "COMPLETE"
```
export enum TaskStatus {
  Complete = 'COMPLETE',
  Failed = 'FAILED',
  InProgress = 'IN_PROGRESS',
}
```

Reply:
But we don't map the SDK task status to the backend anyway, so I can change it. `COMPLETED` is more consistent with `FAILED`

Reply:
Actually now that I'm implementing it, I think we should still keep it consistent with the backend TaskStatus for now. We are building this Enum in hopes that we can map them to the backend TaskStatus, and we're comparing SDK and backend TaskStatus at least for `IN_PROGRESS`. There is then an expectation that all TaskStatuses map correctly. It is not trivial to update the backend TaskStatus either, because SDK 3 customers may depend on it